javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -476,16 +489,24 @@ private Vector(int indexShift, HashArrayMappedTrie<Integer, T> trie) {
 
     @Override
     public Vector<T> append(T element) {
-        return new Vector<>(indexShift, trie.put(length() + indexShift, element));
+        return appendAll(List.of(element));
     }
 
     @Override
     public Vector<T> appendAll(Iterable<? extends T> elements) {
-        HashArrayMappedTrie<Integer, T> result = trie;
+        HashArrayMappedTrie<Integer, T> result = delegate;
+
+        for (int i = offset + length(); i < delegate.size(); i++) {
```

Comment:
in case of append, we delete elements from the end, add the elements back with the correct index and update the length